### PR TITLE
Implement connect timeout & close timeout in Java, remove connection timeouts

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR implement ConnectTimeout and CloseTimeout in Java, and also removes connection timeouts (like already done in C# and C++).

### CODE REVIEW

File: java/test/src/main/java/test/Ice/timeout/AllTests.java [hunk 1]
```diff
@@ -76,28 +53,17 @@ public static void allTestsWithController(test.TestHelper helper, ControllerPrx
...
-
-    int mult = 1;
-    if (!communicator.getProperties().getIceProperty("Ice.Default.Protocol").equals("tcp")
-        || helper.isAndroid()) {
-      mult = 4;
-    }
-
-    TimeoutPrx timeout = TimeoutPrx.checkedCast(obj);
-    test(timeout != null);
+    TimeoutPrx timeout = TimeoutPrx.uncheckedCast(communicator.stringToProxy(sref));
```
Comment (bernardnormier) [inline java/test/src/main/java/test/Ice/timeout/AllTests.java:56]: To be replaced by createProxy.

**Review by pepone — APPROVED** [review]
Looks good, there was a hang with the metrics test on Windows CI not sure if it is related.

File: java/src/Ice/src/main/java/com/zeroc/Ice/ConnectionI.java [hunk 2]
```diff
@@ -1311,15 +1283,10 @@ public ConnectionI(
...
     _timer = instance.timer();
-    _writeTimeout = new TimeoutCallback();
-    _writeTimeoutFuture = null;
-    _readTimeout = new TimeoutCallback();
-    _readTimeoutFuture = null;
     _removeFromFactory = removeFromFactory;
     _warn = initData.properties.getIcePropertyAsInt("Ice.Warn.Connections") > 0;
     _warnUdp =
         instance.initializationData().properties.getIcePropertyAsInt("Ice.Warn.Datagrams") > 0;
-    _cacheBuffers = instance.cacheMessageBuffers();
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply java/src/Ice/src/main/java/com/zeroc/Ice/ConnectionI.java:1322]: It was no longer used.

The previous dispatch pipeline removed all dispatch-side caching.

File: java/test/src/main/java/test/Ice/timeout/AllTests.java [hunk 3]
```diff
@@ -181,80 +117,40 @@ public static void allTestsWithController(test.TestHelper helper, ControllerPrx
...
-      obj.ice_ping();
     }
     out.println("ok");
 
     out.print("testing close timeout... ");
     out.flush();
     {
-      TimeoutPrx to = TimeoutPrx.uncheckedCast(obj.ice_timeout(250 * mult));
-      com.zeroc.Ice.Connection connection = connect(to);
+      var connection = connect(timeout);
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply java/test/src/main/java/test/Ice/timeout/AllTests.java:144]: Removed. Note that this is not a great test - it doesn't test the CloseTimeout.

**Pull request merged**